### DISCUSSION TITLE
SMB file share support

### DISCUSSION THREAD

**Discussion by vandanthaker-ey** [apache/airflow#40474] [Q&A]
I've airflow setup on AKS. I'm switching from NFS to SMB file share under Azure file share. There is a limitation with SMB that it doesn't allow manipulation of permissions within the mounted volume like chown or chmod. Can it be a problem with airflow FileTaskHandler under logging utility, as I see chmod operations getting applied under "_init_file" function?

**Comment by potiuk** [comment]
Yes. Airflow generally assumes a POSIX filesystem that it interects with.

**Reply by vandanthaker-ey** [reply]
Thanks @potiuk  for confirming. although on Azure NFS file share we're facing issues as it doesn't support to use secure transfer - [Link](https://learn.microsoft.com/en-us/azure/storage/files/storage-files-networking-overview#secure-transfer) Which is non negotiable by our security org.
Any other options you could suggest having a persistent volume storage for storing dags, logs, etc.

**Reply by potiuk** [reply]
Change to AWS? or Google? Or use Astronomer ? Also I think you might rise high priority issue to Azure for that and ask what's their suggestion.

**Reply by potiuk** [reply]
Plenty of optiions.

**Reply by vandanthaker-ey** [reply]
Thanks @potiuk but we need to stay within azure. Can this work - https://learn.microsoft.com/en-us/azure/storage/blobs/blobfuse2-what-is ?

**Reply by potiuk** [reply]
No idea.  But you can also use git-sync (no shared volumes at all). This is  what I wrote about a long time ago - that you generally do not need shared volumes at all and in many cases it has a number of problems.  You can read the blog post here https://medium.com/apache-airflow/shared-volumes-in-airflow-the-good-the-bad-and-the-ugly-22e9f681afca and see if it applies to your case, it has a lot of digging in on how things work under the hood, so you might see if the solution you found is good for it.
